libdevice/fallback-bfloat16.cpp

```diff
@@ -0,0 +1,46 @@
+//==------- fallback-bfloat16.cpp - bfloat16 conversions in software -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+
+#include "device.h"
+
+#ifdef __SPIR__
+
+#include <cstdint>
+
+// To support fallback device libraries on-demand loading, please update the
+// DeviceLibFuncMap in llvm/tools/sycl-post-link/sycl-post-link.cpp if you add
+// or remove any item in this file.
+// TODO: generate the DeviceLibFuncMap in sycl-post-link.cpp automatically
+// during the build based on libdevice to avoid manually sync.
+
+DEVICE_EXTERN_C_INLINE uint16_t
+__devicelib_ConvertFToBF16INTEL(const float &a) {
+  // In case float value is nan - propagate bfloat16's qnan
```

Comment:
Why is this code commented off?

Reply:
I think it is because `std::isnan` is not available in device code currently.

Reply:
Yes

Reply:
Regarding necessary changes to llvm-test-suite. They are in the corresponding PR intel/llvm-test-suite#1129.
Cross-testing intel/llvm#5424 and intel/llvm-test-suite#1129 has been successful.

Reply:
Doesn't that mean that the emulation is broken, then?  Is there another way to test for NaN?

Reply:
Can we use a classic `if (a !=a)` here to test if a value is NaN?